### DISCUSSION TITLE
What's the expected duration to generate three sentences using Apple Sillicon chips?

### DISCUSSION THREAD

**Discussion by searene** [neonbjb/tortoise-tts#712] [Q&A]
I have a MacBook Pro with 32GB memory and a m1 chip. I tried generating three sentences with 215 characters, using “fast” preset, and it took 23 minutes. Is it the expected duration? Or maybe I made some mistakes when using tortoise-tts, like not enabling mps? Because it seems very slow for me. I'm not sure if it utilized all the hardware accelerations, I cannot find useful information in the log.

In case it would be helpful, this is the code I used, basically extracted from do_tts.py:

```python
tts = TextToSpeech(models_dir=MODELS_DIR, use_deepspeed=False, kv_cache=True, half=True)

voice_samples, conditioning_latents = load_voices(["halle"])

gen, dbg_state = tts.tts_with_preset(text,
                                     k=3,
                                     voice_samples=voice_samples,
                                     conditioning_latents=conditioning_latents,
                                     # fast, high_quality
                                     preset='fast',
                                     use_deterministic_seed=None,
                                     return_deterministic_state=True,
                                     cvvp_amount=.0)
if isinstance(gen, list):
    for j, g in enumerate(gen):
        torchaudio.save(os.path.join(output_path, f'{output_file_base_name}_{j}.wav'), g.squeeze(0).cpu(), 24000)
else:
    torchaudio.save(os.path.join(output_path, f'{output_file_base_name}.wav'), gen.squeeze(0).cpu(), 24000)
```

**Comment by JMcCoding** [comment]
Not sure if this helps your comparison, but I have a Windows laptop with 16GB, a GeForce GTX 1080, tortoise-tts installed using WSL, and I use deepspeed and kv_cache, but do not use the half flag, and my generations are about 40 minutes per short sentence.  Hopefully someone with a Mac will be able to chime in eventually.

**Edit**: added that I use kv_cache as well.
